docs/topics/spiders.rst

```diff
@@ -259,6 +259,35 @@ to give data more structure you can use :class:`~scrapy.Item` objects::
             for href in response.xpath('//a/@href').getall():
                 yield scrapy.Request(response.urljoin(href), self.parse)
 
+    .. method:: update_settings(settings)
+
+        This is a classmethod which is called to override the configuration of your spider.
+        It receives the settings and passes :attr:`custom_settings` or `{}` to ``setdict`` 
+        method to register the setting with priority ``spider``. 
+        It is preferable to use the :attr:`custom_settings` when making spiders class 
+        hierarchies as parent since :attr:`custom_settings` is shadowed but parent
+        :meth:`update_settings` method can be called.
+
+        :param settings: scrapy settings for the configuration of internal components
+        :type args: dict
+Let's see an example::
+    import scrapy
+    from scrapy.settings import Settings
+    from scrapy.spiders import Spider
+
+
+    class MySpider(scrapy.Spider):
+        spider_class = Spider
+        name = 'example.com'
+        allowed_domains = ['example.com']
+        start_urls = ['http://www.example.com/feed.csv']
+        spider_settings = {'TEST1': 'spider', 'TEST2': 'spider'}
+        project_settings = {'TEST1': 'project', 'TEST3': 'project'}
+        spider_class.custom_settings = spider_settings
+        settings = Settings(project_settings, priority='project')
+        spider_class.update_settings(settings)
```

Comment:
This is the wrong place and indentation (it should be after “This is a classmethod…”, with the same indentation for “Let’s see an example”), and the example is wrong (see https://github.com/scrapy/scrapy/pull/5757 for a better example, yet at the moment also incorrect).